ark/mibi/tiling_utils.py

```diff
@@ -0,0 +1,218 @@
+
+    Returns:
+        dict:
+            Data containing information about each tile, will be saved to JSON
+    """
+
+    # helper function for creating all pairs between two lists
+    def pairs(*lists):
+        for t in combinations(lists, 2):
+            for pair in product(*t):
+                yield pair
+
+    # file path validation
+    if not os.path.exists(moly_path):
+        raise FileNotFoundError("Moly point file %s does not exist" % moly_path)
+
+    # read in the data
+    with open(moly_path, 'r') as mf:
+        moly_point = json.load(mf)
+
+    # get the current time info
+    dt = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S')
+
+    # define the fov tiling info
+    tiled_regions = {
+        'exportDateTime': dt,
+        'fovFormatVersion': tiling_params['fovFormatVersion'],
+        'fovs': []
+    }
+
+    # define a counter to determine where to insert a moly point
+    # only used if tiling_params['moly_interval'] is set
+    moly_counter = 0
+
+    # append tile information to fov_list, iterate through each tiles iteratively
+    for region_index in range(len(tiling_params['fov_num_x'])):
+        # extract start coordinates
+        start_x = tiling_params['region_start_x'][region_index]
+        start_y = tiling_params['region_start_y'][region_index]
+
+        # generate range of x and y coordinates
+        x_range = list(range(tiling_params['fov_num_x'][region_index]))
+        y_range = list(range(tiling_params['fov_num_y'][region_index]))
+
+        x_range_rep = x_range * len(y_range)
+        y_range_rep = y_range * len(x_range)
+
+        # create all pairs between two lists
+        x_y_pairs = list(pairs(x_range, y_range))
+
+        # randomize pairs list if specified
+        if tiling_params['randomize'] == 1:
+            random.shuffle(x_y_pairs)
+
+        for xi, yi in x_y_pairs:
+            # set the current x and y coordinate
+            cur_x = start_x + xi * tiling_params['x_fov_size'][region_index]
+            cur_y = start_y + yi * tiling_params['y_fov_size'][region_index]
+
+            # copy the fov metadata over and add cur_x, cur_y, and identifier
+            fov = copy.deepcopy(tiling_params['fovs'][region_index])
+            fov['centerPointMicrons']['x'] = cur_x
+            fov['centerPointMicrons']['y'] = cur_y
+            fov['name'] = f'row{yi}_col{xi}'
+
+            # append value to tiled_regions
+            tiled_regions['fovs'].append(fov)
+
+            # increment moly_counter as we've added another fov
+            moly_counter += 1
+
+            # append a moly point if moly_interval is set and we've reached the interval threshold
+            if 'moly_interval' in tiling_params and \
+               moly_counter % tiling_params['moly_interval'] == 0:
+                tiled_regions['fovs'].append(moly_point)
+
+        # append moly point to seperate runs if not last
+        if region_index != len(tiling_params['fov_num_x']) - 1:
```

Comment:
This can also be an argument set in previous function that controls whether to insert moly between distinct regions

Reply:
I believe regardless of the other parameters set, moly points always need to be inserted between distinct regions. @efrancis28 can you clarify if this is the case?